free-1.md

```diff
@@ -4,40 +4,130 @@
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+Since this language aims to allow gamers to control units beyond the bounds of
+a typical human reaction time, there must be some coding interface to allow
+users to express their desired behaviors in game. Thus, by allowing users
+to express a controls in code form, they can be sped up to superhuman speeds.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+Although this concept makes for an interesting game idea, the real reason I
+am excited about designing this game is because of its educational value. I
+started playing Starcraft back in 4th grade and its modding tools 
+were my original introduction to programming. Although this game should be
+ideally enjoyable to all programmers, I imagine if the levels are designed
+well with a good progression in difficult, this game could be used to introduce
+completely novice programmers to coding. I've always been amazed with how
+even the most monotonous tasks can be made "fun" in the context of a video
+game by occasionally adding some flashing lights and a "level-up" screen.
+Although I personally think programming is fun to begin with, I think by
+wrapping core coding skills within a enteraining game will help coding
+become more accessable to gamers.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Hardcore gamers teach novice programming.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+As mentioned above, I planned to implement a number of "languages" in the game.
+Originally, I thought making a language that resembled C, Python,
+or JavaScript would be the most "useful" to students since a majority of
+introductory programming classes start with these languages. That being said,
+I have also seen languages like
+[Scratch](https://scratch.mit.edu/) and
+[Blockly](https://developers.google.com/blockly/) which are more targeted
+towards novice coders. In reality, any of these will work and I will probably
+implement some varient of them all.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When a program runs, it is interpreted by the game and executed. Thus, the
+program interacts with the user by carrying out their commands in game. If an
+error occurs in the code, in addition to providing users with a typical error
+message, it would be awesome if a character in the game says something to the
+player like, "Ahh, our onboard computer just segfaulted! Take a look at this
+line of code!" This creates an more personal relationship with the code and
+also adds to the immersive experience. Ideally, we want the users to forget
+they are learning to code!
```

Comment:
I like this. I wonder if it's worth describing to a user what "segfaulted" means.